### PR TITLE
Changes based on consolidation

### PR DESCRIPTION
**Pull request by donaldgray**
Lots of changes:

- Add `JobsController`, `LogsController`, `SettingsController` and move calls from `DashController` over.
- Update Logs views to redirect to Kibana.
- Add `DlcsBasicAuthenticationHandler` to handle auth from DLCS -> RoleProvider.
- Refactor `ArchiveStorageServiceWorkStorageFactory`. Introduced `StorageServiceClient` to avoid needing to pass `ArchiveStorageServiceWorkStorageFactory` to `ArchiveStorageServiceWorkStore` when created.
- Use PostgreSQL for `IDistributedCache`.
- `WellcomeBagAwareArchiveStorageMap`: add static create method. Use Property rather than Fields. Add tests.
- Add docs (readme and sequence diagrams).
- Update `OAuth2ApiConsumer` to automatically refresh access_token and retry once if 403 returned.
- Small fixes based on suggestions (e.g. string interpolation rather than `string.Format()`, use of `async` etc).
- Update `GetStatsModel()` to use PostgreSQL syntax.

### CODE REVIEW

File: src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs [hunk 1]
```diff
@@ -49,13 +46,22 @@ public void ConfigureServices(IServiceCollection services)
...
 
             services.AddMemoryCache();
-            services.AddDistributedMemoryCache();
+            services.AddDistributedPostgreSqlCache(setup =>
+            {
+                setup.ConnectionString = ddsConnectionString;
+                setup.SchemaName = "public";
+                setup.TableName = "__dist_cache";
+                setup.CreateInfrastructure = !WebHostEnvironment.IsProduction();
+                setup.DefaultSlidingExpiration = TimeSpan.FromMinutes(20); // TODO - is this right?
```
Comment (tomcrane) [inline src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs:62]: I think we're setting this explicitly when we use it - how do we set session expiry?

Comment (tomcrane) [reply src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs:62]: That is - is this the same as options.IdleTimeout? very probably not the same.

File: src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs [hunk 2]
```diff
@@ -131,6 +140,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             app.SetupSwagger();
             app.UseStaticFiles();
             app.UseRouting();
+            app.UseAuthorization();
```
Comment (tomcrane) [inline src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs:143]: Does this add any other overhead?

Comment (donaldgray) [reply src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs:143]: Unscientific testing locally there seems to be a slight hit initially but once everything is bootstrapped it's not. Made 4 requests with + without:
```bash
# with
1649ms -> 63ms -> 13ms -> 15ms

# without
1357ms -> 43ms -> 15ms -> 14ms
```

File: src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs (hunk 1 above)
Comment (donaldgray) [reply src/Wellcome.Dds/Wellcome.Dds.Server/Startup.cs:62]: `DefaultSlidingExpiration` and session `IdleTimeout` differ. I had a look at implementation of session and it sets sliding expiry, whereas the above is used if we don't explicity set an expiry.

From dotpeek:

```cs
await _cache.SetAsync(
     _sessionKey,
     stream.ToArray(),
     new DistributedCacheEntryOptions().SetSlidingExpiration(_idleTimeout),
     cts.Token);
```

File: src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Mets/WellcomeBagAwareArchiveStorageMap.cs [hunk 3]
```diff
@@ -10,11 +12,53 @@ namespace Wellcome.Dds.AssetDomainRepositories.Mets
...
-        public List<KeyValuePair<string, HashSet<string>>> VersionSets;
-        public string BucketName;
-        public DateTime StorageManifestCreated;
-        public DateTime Built;
+        /// <summary>
+        /// a List of: "v1" => { "alto/#_0001.xml", ... }
+        /// in increasing order of size of set 
+        /// </summary>
+        /// <remarks>This is a List, as opposed to Dict as ordering is important.</remarks>
+        public List<KeyValuePair<string, HashSet<string>>> VersionSets { get; set; }
```
[the remark this answers is not in the document]
Comment (donaldgray) [reply src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Mets/WellcomeBagAwareArchiveStorageMap.cs:20]: As discussed, had a look and couldn't find anything.

File: src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Mets/StorageServiceClient.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+{
+    public class StorageServiceClient
+    {
+        private readonly OAuth2ApiConsumer oAuth2ApiConsumer;
+        private readonly ILogger<StorageServiceClient> logger;
+        private readonly StorageOptions storageOptions;
+        private readonly ClientCredentials defaultClientCredentials;
+
+        // If this is needed elsewhere it could be better implemented
+        private readonly Dictionary<string, JObject> requestCache = new Dictionary<string, JObject>();
```
[the remark this answers is not in the document]
Comment (donaldgray) [reply src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Mets/StorageServiceClient.cs:20]: Yes - always request scoped.

**Pull request merged**